### PR TITLE
chore(RHTAPWATCH-1043): run Kubelinter on Konflux-ci

### PR DESCRIPTION
**Pull request by ShakedAviv50313**
Running Kubelinter on Konflux-ci

### CODE REVIEW

**Comment by ShakedAviv50313** [comment]
@yftacherzog This pr contains the kube-linter check with all relevant checks disabled.
each disabled check will be enabled one at a time to keep track of the progress of solving issues

**Comment by yftacherzog** [comment]
> @yftacherzog @avi-biton This pr contains the kube-linter check with all relevant checks disabled.
> each disabled check will be enabled one at a time to keep track of the progress of solving issues
>  

Did you confirm that this check is able to fail the ci?

**Comment by ShakedAviv50313** [comment]
I'm not sure what do you mean, each of these checks mentioned in `.kube-linter-config.yaml` is able to fail the ci, this is why they are excluded from the checks. (for now)

File: .github/.kube-linter-config.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+checks:
+  # include explicitly adds checks, by name. You can reference any of the built-in checks.
+  # Note that customChecks defined above are included automatically.
+  include: [ ]
+  # exclude explicitly excludes checks, by name. exclude has the highest priority: if a check is
+  # in exclude, then it is not considered, even if it is in include as well.
+  exclude: [ "unset-cpu-requirements", "unset-memory-requirements", "no-read-only-root-fs", "run-as-non-root", "env-var-secret", "liveness-port",
```
Comment (yftacherzog) [inline .github/.kube-linter-config.yaml]: Can we make it like this (or something similar if I missed something on the syntax)?
```suggestion
  exclude:
    - "unset-cpu-requirements"
    - "unset-memory-requirements"
    - "no-read-only-root-fs"
    - "run-as-non-root"
    - "env-var-secret"
    - "liveness-port"
```

File: .github/workflows/kube-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            dir=$(dirname "$file")
+            dir=${dir#./}
+            output_file=$(echo "out-$dir" | tr "/" "-")
+            kustomize build "$dir" > "./.kube-linter/$output_file.yaml"
+          done
+      - name: Scan yaml files with kube-linter
+        uses: stackrox/kube-linter-action@v1
+        id: kube-linter-action-scan
+        with:
+          # Adjust this directory to the location where your kubernetes resources and helm charts are located.
```
Comment (yftacherzog) [inline .github/workflows/kube-linter.yaml]: Let's remove the comment

File: .github/workflows/kube-linter.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          directory: ./.kube-linter/
+          # Adjust this to the location of kube-linter config you're using, or remove the setting if you'd like to use
+          # the default config.
```
Comment (yftacherzog) [inline .github/workflows/kube-linter.yaml]: Let's remove the comment

**Review by yftacherzog — APPROVED** [review]
lgtm

**Pull request merged**